tools/generator/templates/player/python/interface.cc.jinja2

```diff
@@ -0,0 +1,363 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright (c) 2020 Association Prologin <association@prologin.org>
+{% import 'macros/cxx.jinja2' as cxx %}
+
+// This file contains the code to call the API functions from the Python
+// language.
+// {{ stechec2_generated }}
+
+#include <Python.h>
+#include <vector>
+#include <string>
+
+static PyObject* py_module;
+static PyObject* champ_module;
+
+{% for enum in game.enum %}
+
+{{ cxx.decl_enum(enum) }}
+{% endfor %}
+{% for struct in game.struct %}
+
+{{ cxx.decl_struct(struct) }}
+{% endfor %}
+
+extern "C" {
+{% for func in game.function %}
+
+{{ func.fct_summary|cxx_comment(doc=True) }}
+{{ func|cxx_prototype(prefix='api_') }};
+{% endfor %}
+}
+
+template <typename PythonType, typename CxxType>
+PythonType cxx_to_python(CxxType in)
+{
+    return in.__if_that_triggers_an_error_there_is_a_problem;
+}
+
+template <>
+PyObject* cxx_to_python<PyObject*, int>(int in)
+{
+    return PyLong_FromLong(in);
+}
+
+
+template <>
+PyObject* cxx_to_python<PyObject*, double>(double in)
+{
+    return PyFloat_FromDouble(in);
+}
+
+
+template <>
+PyObject* cxx_to_python<PyObject*, std::string>(std::string in)
+{
+    return PyUnicode_FromString(in.c_str());
+}
+
+
+template <>
+PyObject* cxx_to_python<PyObject*, bool>(bool in)
+{
+    return PyBool_FromLong(in);
+}
+
+template <typename CxxType>
+PyObject* cxx_to_python_array(const std::vector<CxxType>& in)
+{
+    size_t size = in.size();
+    PyObject* out = PyList_New(size);
+
+    for (unsigned int i = 0; i < size; ++i)
+        PyList_SET_ITEM(out, i, (cxx_to_python<PyObject*, CxxType>(in[i])));
+
+    return out;
+}
+
+template <typename PythonType, typename CxxType>
+CxxType python_to_cxx(PythonType in)
+{
+    return in.__if_that_triggers_an_error_there_is_a_problem;
+}
+
+template <>
+int python_to_cxx<PyObject*, int>(PyObject* in)
+{
+    long out = PyLong_AsLong(in);
+    if (PyErr_Occurred())
+        throw 42;
+    return out;
+}
+
+template <>
+double python_to_cxx<PyObject*, double>(PyObject* in)
+{
+    double out = PyFloat_AsDouble(in);
+    if (PyErr_Occurred())
+        throw 42;
+    return out;
+}
+
+template <>
+bool python_to_cxx<PyObject*, bool>(PyObject* in)
+{
+    return (bool) python_to_cxx<PyObject*, int>(in);
+}
+
+template <>
+std::string python_to_cxx<PyObject*, std::string>(PyObject* in)
+{
+    const char * out = PyUnicode_AsUTF8(in);
+    if (PyErr_Occurred())
+        throw 42;
+    return out;
+}
+
+template <typename CxxType>
+std::vector<CxxType> python_to_cxx_array(PyObject* in)
+{
+    if (!PyList_Check(in))
+    {
+        PyErr_SetString(PyExc_TypeError, "a list is required");
+        throw 42;
```

Comment:
Refer to IRC discussion. This sets a PyErr message, so do I understand correctly throwing right after will make Python display a nice traceback (instead of segfaulting)? If not, please see previous comment about printing something debuggable.